### PR TITLE
Avoid ssh keys

### PR DESCRIPTION
**Pull request by whites11**
Towards: https://github.com/giantswarm/giantswarm/issues/13783

This PR check the ssh key fields in azuremachine and azuremachinepool are empty.
We don't use em and don't want the customer to think they can

### CODE REVIEW

File: pkg/azuremachine/validate_create.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	"k8s.io/api/admission/v1beta1"
+	capzv1alpha3 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-admission-controller/internal/errors"
+	"github.com/giantswarm/azure-admission-controller/pkg/validator"
+)
+
+type CreateValidator struct {
+	ctrlClient client.Client
```
Comment (fiunchinho) [inline pkg/azuremachine/validate_create.go:17]: I don't think you are using the client?

File: pkg/azuremachine/validate_update.go [hunk 2]
```diff
@@ -51,6 +51,11 @@ func (a *UpdateValidator) Validate(ctx context.Context, request *v1beta1.Admissi
 		return false, microerror.Maskf(errors.ParsingFailedError, "unable to parse AzureMachine CR: %v", err)
 	}
 
+	err := checkSSHKeyIsEmpty(ctx, AzureMachineNewCR)
```
Comment (fiunchinho) [inline pkg/azuremachine/validate_update.go:54]: I know it's not from this PR but `AzureMachineNewCR` should start with lowercase.

**Review by fiunchinho — COMMENTED** [review]
Would it be possible to add some simple tests?

File: pkg/azuremachine/validate_create.go (hunk 1 above)
Comment (whites11) [reply pkg/azuremachine/validate_create.go]: removed

File: pkg/azuremachine/validate_update.go (hunk 2 above)
Comment (whites11) [reply pkg/azuremachine/validate_update.go:54]: renamed

**Comment by whites11** [comment]
> Would it be possible to add some simple tests?

done

**Review by fiunchinho — APPROVED** [review]
you rock

**Pull request merged**